sentry-ruby/lib/sentry/rack/capture_exceptions.rb

```diff
@@ -1,35 +1,48 @@
 module Sentry
   module Rack
-    class Tracing
+    class CaptureExceptions
       def initialize(app)
         @app = app
       end
 
       def call(env)
-        Sentry.clone_hub_to_current_thread unless Sentry.get_current_hub
-
-        if Sentry.configuration.traces_sample_rate.to_f == 0.0
-          return @app.call(env)
-        end
+        # make sure the current thread has a clean hub
+        Sentry.clone_hub_to_current_thread
 
         Sentry.with_scope do |scope|
           scope.clear_breadcrumbs
```

Comment:
Is this still needed?

Reply:
@matthieuprat it's a way to make sure the request scope doesn't contain any pre-existed data. of course, it can be improved. does this approach cause any issue in your app?

Reply:
It doesn't cause any issue with our app, no.

I was asking because we drew inspiration from that middleware to track errors elswhere in our app and I wasn't sure why this particular line was needed. I thought that maybe it was no longer needed since `clone_hub_to_current_thread` was now being called everytime. But in fact, my understanding from your answer is that we want to discard any breadcrumbs from before the request actually started (which makes sense to me!).

Out of curiosity, why aren't we doing the same thing in the [`ActiveJob` integration](https://github.com/getsentry/sentry-ruby/blob/1bf154e4d9198469df9514821e23e6d65aeffdcb/sentry-rails/lib/sentry/rails/active_job.rb#L11)?

Reply:
@matthieuprat sorry for the delay. it's because I didn't see the same breadcrumb data slipped into AJ events. I didn't dig into the reason though so I may be wrong.

Reply:
No worries! OK, makes sense. (I haven't tested the AJ integration myself so can't tell whether this is an actual issue.)